### PR TITLE
OGD-423 Individuals Matching P1 - PRODUCTION

### PR DESCRIPTION
**Pull request by edgarjimenez**

### CODE REVIEW

File: conf/application.conf [hunk 1]
```diff
@@ -91,19 +91,33 @@ controllers {
...
-    uk.gov.hmrc.individualsmatchingapi.controllers.SandboxPrivilegedCitizenMatchingController = {
+    uk.gov.hmrc.individualsmatchingapi.controllers.MicroserviceHelloWorld = {
         needsLogging = false
         needsAuditing = false
     }
 
-    uk.gov.hmrc.individualsmatchingapi.controllers.SandboxPrivilegedIndividualsController = {
+    #### P1 Live Controllers
+
+    uk.gov.hmrc.individualsmatchingapi.controllers.LivePrivilegedCitizenMatchingController = {
```
Comment (tom59) [inline conf/application.conf]: you can remove LivePrivilegedCitizenMatchingController and LivePrivilegedIndividualsController as they are true by default (I think)

File: project/MicroServiceBuild.scala [hunk 2]
```diff
@@ -9,13 +9,14 @@ object MicroServiceBuild extends Build with MicroService {
   override lazy val appDependencies: Seq[ModuleID] = compile ++ test()
 
   val compile = Seq(
-    "uk.gov.hmrc" %% "play-reactivemongo" % "6.0.0",
+    "uk.gov.hmrc" %% "play-reactivemongo" % "5.2.0",
```
Comment (tom59) [inline project/MicroServiceBuild.scala]: downgrade?

File: project/MicroServiceBuild.scala [hunk 3]
```diff
@@ -9,13 +9,14 @@ object MicroServiceBuild extends Build with MicroService {
...
     ws,
     "uk.gov.hmrc" %% "microservice-bootstrap" % "5.16.0",
     "uk.gov.hmrc" %% "play-authorisation" % "4.3.0",
     "uk.gov.hmrc" %% "play-health" % "2.1.0",
     "uk.gov.hmrc" %% "play-config" % "4.3.0",
     "uk.gov.hmrc" %% "play-hmrc-api" % "1.4.0",
+    "uk.gov.hmrc" %% "play-reactivemongo" % "5.2.0",
```
Comment (tom59) [inline project/MicroServiceBuild.scala]: duplicate?

File: test/unit/uk/gov/hmrc/individualsmatchingapi/controllers/PrivilegedIndividualsControllerSpec.scala [hunk 4]
```diff
@@ -18,82 +18,118 @@ package unit.uk.gov.hmrc.individualsmatchingapi.controllers
...
+      status(eventualResult) mustBe OK
+      contentAsJson(eventualResult) mustBe Json.parse(response(uuid, "Joe", "Bloggs", "AB123456C", "1969-01-15"))
+    }
+
+    "fail with AuthorizedException when the bearer token does not have enrolment read:individuals-matching" in new Setup {
+      given(mockAuthConnector.authorise(refEq(Enrolment("read:individuals-matching")), refEq(EmptyRetrieval))(any())).willReturn(failed(new InsufficientEnrolments()))
+
+      intercept[InsufficientEnrolments] {
+        await(liveController.matchedIndividual(uuid.toString).apply(FakeRequest()))
+      }
```
Comment (tom59) [inline test/unit/uk/gov/hmrc/individualsmatchingapi/controllers/PrivilegedIndividualsControllerSpec.scala]: add verifyZeroInteraction(mockCitizenMatchingService)

File: test/unit/uk/gov/hmrc/individualsmatchingapi/controllers/PrivilegedCitizenMatchingControllerSpec.scala [hunk 5]
```diff
@@ -117,30 +129,103 @@ class PrivilegedCitizenMatchingControllerSpec extends UnitSpec with MockitoSugar
...
 
-    "not require bearer token authentication" in new Setup {
-      when(mockSandboxCitizenMatchingService.matchCitizen(any[CitizenMatchingRequest])(any[HeaderCarrier]))
-        .thenReturn(Future.successful(matchId))
+    "fail with AuthorizedException when the bearer token does not have enrollment read:individuals-matching" in new Setup {
+      var requestBody = Json.parse("""{"firstName":"Amanda","lastName":"Joseph","nino":"NA000799C","dateOfBirth":"2020-01-32"}""")
+
+      given(mockAuthConnector.authorise(refEq(Enrolment("read:individuals-matching")), refEq(EmptyRetrieval))(any())).willReturn(failed(new InsufficientEnrolments()))
+
+      intercept[InsufficientEnrolments]{await(liveController.matchCitizen()(fakeRequest.withBody(requestBody)))}
```
Comment (tom59) [inline test/unit/uk/gov/hmrc/individualsmatchingapi/controllers/PrivilegedCitizenMatchingControllerSpec.scala]: add verifyZeroInteraction(mockLiveCitizenMatchingService)

File: project/MicroServiceBuild.scala (hunk 2 above)
Comment (edgarjimenez) [reply project/MicroServiceBuild.scala]: Didn't work first-time with the new version, so I reverted and then forgot. Can revisit...

**Pull request merged**